net-wireless/rpi3-bluetooth/files/1.2/rpi3-bluetooth.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Starts up integrated bluetooth on the RPi3"
+After=bluetooth.service
+
+[Service]
+ExecStart=/usr/lib/rpi-scripts/bin/rpi3-bluetooth start
+ExecStop=/usr/lib/rpi-scripts/bin/rpi3-bluetooth stop
+Type=forking
```

Comment:
I recommend against using the "Type=forking" model of systemd services.

Specifically in this service the only reason to use "forking" is because the actual service is started by "nohup".

Instead of calling the rpi3-bluetooth script, just call /usr/sbin/rpi3-attach-bluetooth directly.

Systemd also directly supports requiring specific paths to exist in sysfs or proc before starting a service, so you can prevent this service from starting if /proc/device-tree/soc/gpio@7e200000/bt_pins isn't found.

Reply:
Part of the reason I did it that way was so that the same script could be used from openrc or systemd.
The idea being that it would be more likely to be accepted if it could be more easily maintained.